### ISSUE TITLE
Inconsistent Handling of gguf_writer.add_name in convert-to-gguf.py Scripts

### ISSUE THREAD

**Issue by maddes8cht** [ggerganov/llama.cpp#3775]
# short
* gguf_writer.add_name is handled  inconsistent along the different convert-scripts
* let's clarify the intended content of gguf_writer.add_name
* Define a consistent approach for setting the name in the gguf_writer.add_name
* Consider allowing for a commandline override of the name through a command-line option.

# long
gguf_writer.add_name is handled in different ways in the convert-to-gguf.py scripts:

In `convert-llama-ggml-to-gguf.py` there is this block of code, which is processed for the name to be determined:
```
        hp = self.model.hyperparameters
        cfg = self.cfg
        if cfg.desc is not None:
            desc = cfg.desc
        else:
            desc = f'converted from legacy {self.model.file_format.name}v{self.model.format_version} {hp.ftype.name} format'
        try:
            # Filenames aren't necessarily valid UTF8.
            name = cfg.name if cfg.name is not None else cfg.input.name
        except UnicodeDecodeError:
            name = None
        print('* Adding model parameters and KV items')
        if name is not None:
            gguf_writer.add_name(name)
```
In `convert-falcon-hf-to-gguf.py`, however, it simply says
```
gguf_writer.add_name("Falcon")
```
in `convert-bloom-hf-to-gguf.py` accordingly
```
gguf_writer.add_name("Bloom")
```
which still makes sense, in `convert-persimmon-to-gguf.py` it says 
```
gguf_writer.add_name('persimmon-8b-chat')
```
which doesn't make sense anymore, because not only persimmon-8b-chat should be converted, but also persimmon-8b-base or derived models.

In most of the other convert-scripts you will find in the meantime
```
gguf_writer.add_name(dir_model.name)
```
which corresponds to the directory name where the model to be converted is located.

The idea behind this is obviously that it is the directory name that is generated from huggingface in a simple git clone.
But this assumption is deceptive and would have required at least some documentation.
So far I always load the source models into the same directory and quantize only at the end into a directory with a speaking name. Therefore I also came across this circumstance, because in all models with this procedure in the convert script with me the same name in the gguf file stands.

Basically the `name` should be able to contain a different information than the model type - in this respect a repetition like currently in falcon and Bloom is not very helpful.
If it is generally agreed to enter the dir_model.name by default, an override by means of a commandline option would be helpful, though.

A unification of the scripts would be very helpful at this point and I can gladly write a PR about it, but would like to clarify first which direction can meet general approval.
